### DISCUSSION TITLE
Help with Registering an Existing Linux Server in Coder

### DISCUSSION THREAD

**Discussion by Suchun-sv** [coder/coder#14339] [General]
Hi,

Thank you for your excellent work!

I'm trying to manage all my resources—Docker, VMs, and remote Linux machines—using Coder. However, I'm having trouble registering an existing Linux server with Coder.

What I'd like is to be able to click a "VS Code" button in Coder and connect directly to this Linux server, similar to how the VS Code SSH plugin works. Please note that this server doesn't have Docker or sudo permissions. 

Could you guide me on how to achieve this?

Best regards,
Suchunsv

**Comment by kylecarbs** [comment]
Is this for an ephemeral environment or are you just looking to connect it? If looking to connect a VM, I might suggest just using code-server instead. It would be simpler!

**Comment by matifali** [comment]
A related discussion is here: https://github.com/coder/coder/issues/10121

**Comment by matifali** [comment]
@Suchun-sv have you looked into using the https://registry.terraform.io/providers/loafoe/ssh/latest/docs
It allows you to run commands over ssh in an existing server/VM.
